### PR TITLE
Add React and rewrite script.js to use it

### PR DESCRIPTION
**Pull request by n-wach**

### CODE REVIEW

**Comment by n-wach** [comment]
Actually going to draft this until #45 and #46 are approved and merged

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -59,224 +59,178 @@ function openProject() {
...
+    return (<div class={CLASS_COMMENT}>
+        <div class={CLASS_COMMENT_TOPLINE}>
+          <span>{author} on {comment.created}</span>
+        </div>
+        <p>{comment.content}</p>
+        { this.props.displayBaseline && <div class={CLASS_COMMENT_BASELINE}>
+          <button>reply</button>
+          <button>report</button>
+        </div>}
+      </div>);
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:77]: The formatting is off here for this piece. Lets start with trying to make sure the braces and parenthesis have appropriate new lines.
```
(
  {
  }
)
```
Sample to check: https://github.com/google-map-react/google-map-react-examples/blob/master/src/components/AutoComplete.js#L56

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -59,224 +59,178 @@ function openProject() {
...
-
-    // TODO: Make these buttons work.
-    const reply = document.createElement('button');
-    reply.innerText = 'reply';
-    baseline.appendChild(reply);
-
-    const report = document.createElement('button');
-    report.innerText = 'report';
-    baseline.appendChild(report);
+class Comment extends React.Component {
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:64]: nit: For the purpose of next time lets make sure that the Reach components are in their own js file and not in a big script file.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -59,224 +59,178 @@ function openProject() {
...
+class CommentSubmission extends React.Component {
+  render() {
+    return (<form action={COMMENTS_ENDPOINT} method='POST'>
+        <input type='hidden' name={PARAM_TOPIC} value={this.props.topic}/>
+        <textarea name={PARAM_CONTENT} placeholder='Type something...'/>
+        <input type='submit' value='Submit'/>
+        <span>as {USER_INFO.user.value.nickname}</span>
+        <a href={USER_INFO.logoutUrl.value}>logout</a>
+      </form>);
+  }
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:94]: ditto for formatting and nit.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -59,224 +59,178 @@ function openProject() {
...
-  login.innerText = "login";
-  div.appendChild(login);
-
-  return div;
+class CommentLogin extends React.Component {
+  render() {
+  return (<div>
+      <h2>Please log in to post a comment</h2>
+      <a href={USER_INFO.loginUrl.value}>login</a>
+    </div>);
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:106]: ditto

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -59,224 +59,178 @@ function openProject() {
...
+    }
 
-  return form;
+    return (<form action={USER_ENDPOINT} method='POST'>
+        <h2>Please set your nickname</h2>
+        <input onInput={this.verifyNickname.bind(this)} name={PARAM_NICKNAME}/>
+        <input type='submit' value={message} disabled={disabled}/>
+        <a href={USER_INFO.logoutUrl.value}>logout</a>
+      </form>);
+  }
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:145]: ditto

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (n-wach) [reply portfolio/src/main/webapp/script.js]: Reformatted

**Comment by n-wach** [comment]
Merging without approval because this project is long over, and don't want these to linger/waste anyone's time.

**Pull request merged**